aws/resource_aws_ssm_resource_data_sync_test.go

```diff
@@ -0,0 +1,112 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/hashicorp/terraform/helper/acctest"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSSsmResourceDataSync_basic(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSsmResourceDataSyncDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSsmResourceDataSyncConfig(acctest.RandInt(), acctest.RandString(5)),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSSsmResourceDataSyncExists("aws_ssm_resource_data_sync.foo"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAWSSsmResourceDataSyncDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).ssmconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_ssm_resource_data_sync" {
+			continue
+		}
+		_, err := findResourceDataSyncItem(conn, rs.Primary.Attributes["name"])
+		if err != nil {
+			if _, ok := err.(awserr.Error); ok {
+				return err
+			} else {
+				return nil
+			}
+		} else {
+			return fmt.Errorf("Resource Data Sync (%s) found", rs.Primary.Attributes["name"])
+		}
+	}
+	return nil
+}
+
+func testAccCheckAWSSsmResourceDataSyncExists(name string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		log.Println(s.RootModule().Resources)
+		_, ok := s.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("Not found: %s", name)
+		}
+		return nil
+	}
+}
+
+func testAccSsmResourceDataSyncConfig(rInt int, rName string) string {
+	return fmt.Sprintf(`
+    resource "aws_s3_bucket" "hoge" {
+      bucket = "tf-test-bucket-%d"
+      region = "us-east-1"
```

Comment:
Any particular reason why this bucket should be in `us-east-1`? It's certainly no big deal, just that our primary test region is `us-west-2`.